.github/actions/setup-project/action.yml

```diff
@@ -0,0 +1,75 @@
+name: Setup Novu Monorepo
+
+description: Sets up the whole monorepo and install dependencies
+
+inputs:
+  cypress_version:
+    description: 'The version of cypress to install'
+    required: false
+    default: ''
+outputs:
+  cypress_cache_hit:
+    description: 'Did cypress used binary cache'
+    value: ${{ inputs.cypress_version != '' && steps.cache-cypress-binary-version.outputs.cache-hit || steps.cache-cypress-binary.outputs.cache-hit}}
+
+runs:
+  using: composite
+  steps:
+    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+    - name: ⚙️ Setup kernel for react native, increase watchers
+      shell: bash
+      run: echo fs.inotify.max_user_watches=524288 | sudo tee -a /etc/sysctl.conf && sudo sysctl -p
+
+    - uses: actions/setup-node@v2
+      name: ⚙️ Setup Node Version
+      with:
+        node-version: '16.15.1'
```

Comment:
In the future we might want to use this GH action:
https://github.com/marketplace/actions/get-nodejs-version

We could also write one for the MongoDB and PNPM versions.

Reply:
that's cool! Love it